Find a functionally equivalent code:com.android.internal.location.GpsNetInitiatedHandler.getDlgIntent:COMMENT
Method Modifier: private     internal    
Comment:// notification and takes user input

@@ -3,9 +3,9 @@
     String title = getDialogTitle(notif, mContext);
     String message = getDialogMessage(notif, mContext);
     // directly bring up the NI activity
-    intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
+    intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TASK);
     intent.setClass(mContext, com.android.internal.app.NetInitiatedActivity.class);
     // put data in the intent
     intent.putExtra(NI_INTENT_KEY_NOTIF_ID, notif.notificationId);

